### ISSUE TITLE
Halbe Wagen

### ISSUE THREAD

**Issue by bensteUEM** [nFrechen/DB-Wagenreihungsplan-XML-to-HTML#1]
Hi, super Idee mit den XSLTs 👍 
Aber zwei Frage:
* ist das Projekt noch Aktiv
* 2. gibts schon Updates zu den halben (bzw. noch gar nicht aufgefallen 2/3 Wagen z.B. beim TGV)

Hätte evtl. Java Code der einige der Probleme - vor allem mit den normalen halben Wagen und der Abschnittszuordnung schon halbwegs löst.

**Comment by nFrechen** [comment]
Hey!

Sorry, wollte dich jetzt nicht so lange warten lassen. Ich fänd's auf jeden Fall cool, das Projekt noch ein wenig weiter zu entwickeln. Gibt ja noch so einiges zu tun. 

Für mich ist XSLT auch noch neu, von daher bin ich auf manche Sachen auch noch nicht gekommen. Z.B. ob man das Template auch auf eine XML-Datei anwenden kann ohne sie im Header der XML-Datei als Template zu deklarieren. Vielleicht könnte man da was mit JavaScript machen. Auf diese Weise könnte man die XML-Dateien dann direkt von der Quelle rein laden.

Manche Sachen, wie das mit dem TGV, muss man entweder aus anderen Quellen wissen oder noch etwas über die Daten meditieren... Ich mein der ICE Steuerstand ist ja jetzt auch nicht wirklich zur Hälfte Steuerstand, sondern eher 10% der Länge oder so.

Ich wollte eigentlich auch nochmal ne Mail an die von DB OpenData schicken. Da müsste ich meine Fragen aber denke ich noch etwas konkretisieren...

Wenn du was beisteuern möchtest gerne!

**Comment by bensteUEM** [comment]
ok gut zu wissen - genau das mit dem Meditieren meinte ich :+1: Wird vermutlich noch etwas dauern, aber ich gucke mal das ich den Code auch irgendwie nach Github bekomme, ist wohl Java für eine Android Applikation, mit XSLTs habe ich noch nicht wirklich viel gemacht

**Comment by nFrechen** [comment]
Achso, Java oder Java Script?

Um hier auf Github was beizutragen müsstest du das Projekt auf deinen eigenen Account forken, ändern und dann einen Pull-Request machen. Kann dir gerne zeigen, wie das geht.

**Comment by bensteUEM** [comment]
@nFrechen - ja ich weiß wie github funktioniert, habe das nur zzt in nem privaten repository mit nen paar Sachen gemischt :)

Und ja es ist Java, es geht mir aber mehr um die Logik statt ums copy paste vom Code, wenn man einmal interpretiert hat wie die Wagen in welche Section gehören ist das ja unabhängig von der Programmiersprache

**Comment by nFrechen** [comment]
Sind die Abschnitte denn in allen Bahnhöfen gleich lang? Passen also immer gleich viele Waggons rein?

Man könnte vielleicht sozuagen einen Standardbahnsteig machen und dann versucht man die Züge bestmöglich darüber zu schieben (sozusagen eine Optimierungsaufgabe): 
```
Bahnsteig: A, A, A, A, A, B, B, C, C, D, D, E, E, E, E, E
Zug1:               A, A, B, B, C, C, D, D, E, E
Zug2:            A, A, A, B, B, C, C, D, D
```

**Comment by bensteUEM** [comment]
Ja, die Abschnitte sind ca. gleich lang, bin leider noch nicht dazu gekommen den anderen Code aufzuräumen, steht aber auf meiner ToDo ...
